framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1819,6 +1820,11 @@ protected void checkPreconditions(MethodInvocationTree tree, Set<Precondition> p
         QualifierHierarchy hierarchy = atypeFactory.getQualifierHierarchy();
         Set<AnnotationMirror> annos = value.getAnnotations();
         inferredAnno = hierarchy.findAnnotationInSameHierarchy(annos, anno);
+      } else if (exprJe instanceof FieldAccess) {
```

Comment:
I think this could just be `} else {`.  I don't think it matters that the expression is a FieldAccess.